File: src/plugins/autostart.ts
import { invoke } from "@tauri-apps/api/core";

const COMMAND = {
  IS_AUTOSTART: "plugin:eco-autostart|is_autostart",
};

/**
 * 是否为开机自动启动
 */
export const isAutostart = () => {
  return invoke<boolean>(COMMAND.IS_AUTOSTART);
};

File: src-tauri/src/commands/autostart.rs
use tauri::AppHandle;

use crate::autostart;
use crate::core::Result;

#[tauri::command]
pub async fn get_autostart(app: AppHandle) -> Result<bool> {
    autostart::is_enabled(&app)
}

#[tauri::command]
pub async fn set_autostart(app: AppHandle, enabled: bool) -> Result<()> {
    autostart::set_enabled(&app, enabled)
}

#[tauri::command]
pub async fn is_launched_via_autostart() -> bool {
    autostart::launched_via_autostart()
}
